### ISSUE TITLE
ngSanitize removes HTML strings between script elements

### ISSUE THREAD

**Issue by memolog** [angular/angular.js#11442]
ngSanitize removes HTML strings between scripts elements. If I have the following HTML strings, the expected result is 'ABC' but the actual is 'AC'. The B is gone.

```
A
<script>evil</script>
B
<script>evil</script>
C
```
Plunker
http://plnkr.co/lQcfIh

After finding the begin of the script tag, ngSanitize searches the string of the end tag, like ```</script>``` with the longest match, so it removes all of the HTML strings between script elements. I think we should use the smallest possible match.

I will send the PR for it later. Thanks in advance!

----
Angular Version(s) : 1.4.0-beta
Browsers and Operating System : Chrome 41 / Mac OS 10.8.5

**Comment by memolog** [comment]
I found my PR didn't pass the following test, but I think it's acceptable because its purpose is checking that the script tag is not executable. The rest of html strings will go to the next while loop, so they will be sanitized properly.

```
Running "tests:jqlite" (tests) task
INFO [karma]: Karma v0.12.32 server started at http://localhost:9876/
INFO [launcher]: Starting browser Chrome
INFO [Chrome 41.0.2272 (Mac OS X 10.8.5)]: Connected on socket rRl8EsKEWtkVNu9CAAAA with id 90376227
Chrome 41.0.2272 (Mac OS X 10.8.5) HTML should remove double nested script FAILED
	Expected 'ailc.' to equal 'ac.'.
	Error: Expected 'ailc.' to equal 'ac.'.
	    at null.<anonymous> (/path/to/angular.js/test/ngSanitize/sanitizeSpec.js:176:65)
Chrome 41.0.2272 (Mac OS X 10.8.5): Executed 4036 of 4036 (1 FAILED) (18.829 secs / 18.431 secs)
```

**Comment by petebacondarwin** [comment]
I think we should be able to fix this with #11790
